apps/web/src/state/farms/hooks.ts

```diff
@@ -95,38 +91,28 @@ export const usePollFarmsAvgInfo = (activeFarms: (V3FarmWithoutStakedValue | V2F
 
       const addresses = activeFarms.map((farm) => farm.lpAddress?.toLowerCase())
 
-      const rawResult: any | undefined = await multiQuery(
-        (subqueries) => gql`
-      query getVolume {
-        ${subqueries}
+      const [v2AvgRes, v3AvgRes, stableAvgRes] = await Promise.allSettled([
+        fetchV2FarmsAvgInfo(chainId),
+        fetchV3FarmsAvgInfo(chainId),
+        fetchStableFarmsAvgInfo(chainId),
+      ])
+      const farmAvgInfo = {
+        ...(v2AvgRes.status === 'fulfilled' ? v2AvgRes.value : {}),
+        ...(v3AvgRes.status === 'fulfilled' ? v3AvgRes.value : {}),
+        ...(stableAvgRes.status === 'fulfilled' ? stableAvgRes.value : {}),
       }
-    `,
-        addresses.map(
-          (tokenAddress) => `
-          t${tokenAddress}:poolDayDatas(first: 7, orderBy: date, orderDirection: desc, where: { pool: "${tokenAddress.toLowerCase()}"}) {
-            volumeUSD
-            tvlUSD
-            feesUSD
-            protocolFeesUSD
-          }
-    `,
-        ),
-        client,
-      )
-
-      const results = mapKeys(rawResult, (_, key) => key.substring(1, key.length))
-
-      return mapValues(results, (value) => {
-        const volumes = value.map((d: { volumeUSD: string }) => Number(d.volumeUSD))
-        const feeUSDs = value.map(
-          (d: { feesUSD: string; protocolFeesUSD: string }) => Number(d.feesUSD) - Number(d.protocolFeesUSD),
-        )
-        return {
-          volumeUSD: averageArray(volumes),
-          tvlUSD: parseFloat(value[0]?.tvlUSD) || 0,
-          feeUSD: averageArray(feeUSDs),
+
+      const info: { [key: string]: { volumeUSD: number; tvlUSD: number; feeUSD: number; apr: number } } = {}
+      for (const addr of addresses) {
+        const farmInfo = farmAvgInfo[addr]
+        info[addr] = {
+          volumeUSD: farmInfo?.volumeUSD7d.decimalPlaces(2).toNumber(),
```

Comment:
The original value is avg 7d right?

Reply:
yes correct. It was using 7day value to calculate